modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +612,77 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  context('dispatchChange', () => {
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevelContent = 'a modified last level';
+    let editor: Editor;
+
+    const assertChangeEvent = (
+      eventContent: { levelContent: string; lastLevelContent: string | undefined },
+      expectedLevelContent: string | undefined,
+      expectedLastlevelContent: string | undefined
+    ) => {
+      assert.deepEqual(eventContent, {
+        levelContent: expectedLevelContent,
+        lastLevelContent: expectedLastlevelContent
+      }, 'the level or the last level have not the expected content');
+    };
+
+    let changeEventCounter: number;
+    let currentChangeEvent: { levelContent: string; lastLevelContent: string } | undefined;
+
+    const onChange = (e: EditorEvent<{
+      level: UndoLevel;
+      lastLevel: UndoLevel | undefined;
+    }>) => {
+      changeEventCounter++;
+      currentChangeEvent = { levelContent: e.level?.content, lastLevelContent: e.lastLevel?.content };
+    };
+
+    beforeEach(() => {
+      changeEventCounter = 0;
+      currentChangeEvent = undefined;
+
+      editor = hook.editor();
+      editor.resetContent(initialContent);
+      editor.on('change', onChange);
+    });
+
+    it('TINY-8641: Dispatch change with current editor status as level and current undoManager layer as lastLevel', () => {
+      assert.equal(changeEventCounter, 0, 'No events should be detected at start');
+
+      Arr.last(editor.undoManager.data).each((lastLevel) => {
+        lastLevel.content = manualModifiedLevelContent;
+      });
+
+      assert.isFalse(editor.isDirty(), 'Editor should not be dirty before dispatchChange');
+      editor.undoManager.dispatchChange();
+      assertChangeEvent(currentChangeEvent, initialContent, manualModifiedLevelContent);
+      assert.equal(changeEventCounter, 1, '1 event should be detected');
+      assert.isTrue(editor.isDirty(), 'Editor should be dirty after dispatchChange');
+
+      Arr.range(3, () => {
+        editor.undoManager.dispatchChange();
+        assertChangeEvent(currentChangeEvent, initialContent, manualModifiedLevelContent);
+      });
+
+      assert.equal(changeEventCounter, 4, 'it should continue to call change');
```

Comment:
Does this really have any benefit now that we want it to fire at all times anyways? It was useful for when we wanted it to skip, but this doesn't really seem to be testing anything meaningful now.

It looks to me at least that all of this could be condensed down to a single `it` test.

Reply:
ok, I removed the loops but I think that have 2 tests make sense since the second one cover an edge case